### DISCUSSION TITLE
ComboBox which is filterable and multiselect capable?

### DISCUSSION THREAD

**Discussion by tblong** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3354] [Q&A] [answered]
I have a hunch this is out of scope for material (since nothing seems to exist in the spec: https://m2.material.io/components/menus#dropdown-menu) in general but figured I'd ask anyways in case someone had some helpful suggestions on which direction I should take.

**The need**: Looking for a combobox which is both multi-select and filter capable similar to the web based example below.

**Example**: 
https://carbondesignsystem.com/components/dropdown/usage/#live-demo
![image](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/12371929/6123512e-edf8-4ed9-b6b2-0d3698dc5780)

**Questions**:
1. Is this completely out of scope in a strictly material world?
2. How should one go about creating their own from the perspective of a noob who has only used this material project and never had to create their own custom control?

**Comment by Keboo** [comment, marked as answer]
It is an interesting idea that I have seen asked for a few times. 
Personally if I were to go build something, I would not try to make the ComboBox work and instead put it together from its individual pieces. 
A ItemsSource with a WrapPanel for the top section that displays selection. I would do an ItemTemplate to get that pill look with the X to remove. A ToggleButton styled to look like a combo box arrow. A Popup with its IsOpen property bound to the toggle button. And an ItemsSource with a ScrollViewer inside of the popup to display the individual items. Then it would be a matter of getting things all wired. 

The image seems to imply the user can type in the top section but I would not do that. If some sort of search/filter is desired I would put that in the popup directly rather than trying to overload the top section.

**Reply by tblong** [reply]
Is it safe to presume this is out of scope to incorporate into this material project?  I'm failing to find any spec'd MD component which outlines a searchable/filterable dropdown.  If the new control described here is worthy happy to help contribute a first go at it.  Still grateful for the guidance either way; its time I get comfortable building my own controls anyways.

**Reply by Keboo** [reply]
At least in the near future yes, it is out of scope. The intent of this library is to be a theme library. We have been dabbling in providing some custom controls on a limited basis when they are needed to support broad use-cases (and there have been some request for similar things before). But at present there are no plans to bring in a control quite like this.
